### PR TITLE
feat: Add django-post_office for mail queueing

### PR DESCRIPTION
**Pull request by sakhawy**
#312

### CODE REVIEW

**Comment by sarahboyce** [comment]
Sorry Moe I just merged a dependabot so this needs a rebase/conflict resolution ❤️ 

This looks good, we either need celery or something similar set up (https://github.com/djangonaut-space/wagtail-indymeet/issues/270) or a cron job (https://github.com/ui/django-post_office?tab=readme-ov-file#quickstart) in order to send these emails.
I think we haven't quite figured out how to do this yet 🤔 (cc: @dawnwages @tim-schilling)

**Comment by sakhawy** [comment]
I tried to hack my way through with ``supervisord`` :smile: 

```
[program:post_office_send_queued_mail]
command=bash -c 'python /app/manage.py send_queued_mail && sleep 30'
autorestart=true

[program:post_office_cleanup_mail]
command=bash -c 'python /app/manage.py python manage.py cleanup_mail --days=30 --delete-attachments && sleep 86400'
autorestart=true
```

**Comment by tim-schilling** [comment]
@sakhawy how much testing of that hack did you do? Is it something we can merge and go, or do I need to do some verification when it gets pushed to staging?

**Comment by sakhawy** [comment]
Here's how I tested it:
- `docker compose up`
- `docker exec -it <wagtail-indymeet-web-container-id> bash`
- `supervisord -c supervisord.conf`
- I redirected post_office management command's stdout & stderr to a log file in supervisord.conf
- The log file was sending mail every 30 seconds

I'd say some verification is needed just to make sure.

File: indymeet/settings/base.py [hunk 1]
```diff
@@ -222,6 +223,14 @@
 SERVER_EMAIL = "contact@djangonaut.space"
 SILENCED_SYSTEM_CHECKS = ["django_recaptcha.recaptcha_test_key_error"]
 
+EMAIL_BACKEND = "post_office.EmailBackend"
+
+POST_OFFICE = {
+    "BACKENDS": {
+        "default": "django.core.mail.backends.console.EmailBackend",
```
Comment (tim-schilling) [inline indymeet/settings/base.py:230]:
```suggestion
        "default": "anymail.backends.mailjet.EmailBackend",
```

**Comment by tim-schilling** [comment]
I changed it to default the production setting in base.py and override in dev rather than production.

**Comment by tim-schilling** [comment]
Thanks @sakhawy!

**Comment by sarahboyce** [comment]
We're no longer sending emails on staging

@sakhawy if it helps, the docker configuration is just to aid local setup. We're not deploying this image

**Comment by sakhawy** [comment]
Thanks @sarahboyce! Yeah, I was just trying to emulate supervisord running management commands periodically on startup.

**Pull request merged**